crates/nargo_cli/tests/test_data/brillig_not/src/main.nr

```diff
@@ -2,8 +2,8 @@
 // 
 // The features being tested is not instruction on brillig
 fn main(x: Field, y : Field)  {
-    assert(false == not_operator(x as bool));
-    assert(true == not_operator(y as bool));
+    assert(false == not_operator(x == 1));
+    assert(true == not_operator(y == 1));
```

Comment:
Why were the `as bool` casts removed?

Reply:
The workaround in truncate_var does not work for u1, but in fact such cast was not needed in the test suite.
I believe this is OK until we can re-write this part when we have bigints.

Reply:
I implemented a much better alternative which does not require the workaround and works with `as bool `castings.